sql/log_event.cc

```diff
@@ -10850,6 +10850,7 @@ static enum_tbl_map_status check_table_map(Relay_log_info const *rli,
 #ifdef WITH_WSREP
   if ((rli->info_thd->slave_thread /* filtering is for slave only */ ||
        (WSREP(rli->info_thd) && rli->info_thd->wsrep_applier)) &&
+      wsrep_check_mode(WSREP_MODE_IGNORE_NATIVE_REPLICATION_FILTER_RULES) &&
```

Comment:
Shouldn't we stop checking further and return from the function when `WSREP_MODE_IGNORE_NATIVE_REPLICATION_FILTER_RULES` is set?

Please note that, in 5.7 we have the same condition as 
```
#ifdef WITH_WSREP
  /* Check if we should ignore replication filter rules for cluster events. */
  const bool thd_is_wsrep_applier=
    ┆ WSREP(rli->info_thd) && rli->info_thd->wsrep_applier;
  if (thd_is_wsrep_applier &&
      wsrep_check_mode(WSREP_MODE_IGNORE_NATIVE_REPLICATION_FILTER_RULES))
    DBUG_RETURN(res);

  if ((rli->info_thd->slave_thread /* filtering is for slave only */ ||
    ┆  thd_is_wsrep_applier) &&
#else
  if (rli->info_thd->slave_thread /* filtering is for slave only */ &&
#endif /* WITH_WSREP */
```

Reply:
I made a mistake when I copied this line from one of my checkouts to another, fixed now. Now it looks the same as in upstream codership.

Reply:
Looks like this logic doesn't do well with native replication filters.
We have these rpl tests failing with this change.
```
[ 34%] rpl.rpl_filter_with_default_channel 'row' w3  [ fail ]
[ 39%] rpl.rpl_filters_check_counter 'row'      w3  [ fail ]
[ 60%] rpl.rpl_binlog_transaction_dependency_tracking_with_filters 'row' w3  [ fail ]
[ 68%] rpl.rpl_do_db_per_channel_filter 'row'   w3  [ fail ]
[ 72%] rpl.rpl_do_table_per_channel_filter 'row' w4  [ fail ]
[ 76%] rpl.rpl_ignore_db_per_channel_filter 'row' w1  [ fail ]
[ 83%] rpl.rpl_ignore_table_per_channel_filter 'row' w3  [ fail ]
[ 91%] rpl.rpl_wild_do_table_per_channel_filter 'row' w1  [ fail ]
[ 93%] rpl.rpl_wild_ignore_table_per_channel_filter 'row' w3  [ fail ]
```

Reply:
The following tests are also failing. Please fix them.
1. main.mysqld--help-notwin
2. all_persisted_variables